src/main/java/spliterators/part4/data/Node.java

```diff
@@ -28,8 +30,6 @@ public T getValue() {
     }
 
     public Stream<T> stream() {
-        // TODO
-        throw new UnsupportedOperationException();
-        //return StreamSupport.stream(new NodeSplitertor(this));
+        return StreamSupport.stream(new NodeSpliterator<>(this), true);
```

Comment:
`stream()` should return `sequential` `Stream`. You could add `parallelStream()`.